adafruit_bno080/uart.py

```diff
@@ -0,0 +1,163 @@
+# SPDX-FileCopyrightText: Copyright (c) 2020 Bryan Siepert for Adafruit Industries
+#
+# SPDX-License-Identifier: MIT
+"""
+
+    Subclass of `adafruit_bno080.BNO080` to use UART
+
+"""
+import time
+from struct import pack_into
+from . import BNO080, BNO_CHANNEL_EXE, BNO_CHANNEL_SHTP_COMMAND, DATA_BUFFER_SIZE, const, Packet, PacketError
+
+
+class BNO080_UART(BNO080):
+    """Library for the BNO080 IMU from Hillcrest Laboratories
+
+        :param uart: The UART devce the BNO080 is connected to.
+
+    """
+
+    def __init__(self, uart, reset=None, debug=False):
+        self._uart = uart
+        super().__init__(reset, debug)
+
+    def _send_packet(self, channel, data):
+        data_length = len(data)
+        write_length = data_length + 4
+        ba = bytearray(1)
+        
+        # request available size
+        """
+        self._uart.write(b'\x7e') # start byte
+        time.sleep(0.001)
+        self._uart.write(b'\x00') # SHTP byte
+        time.sleep(0.001)
+        self._uart.write(b'\x7e')
+        avail = self._uart.read(5)
+        if avail[0] != 0x7e and avail[1] != 0x01 and avail[4] != 0x7e:
+            raise RuntimeError("Couldn't get available buffer size")
+        """
+            
+        pack_into("<H", self._data_buffer, 0, write_length)
+        self._data_buffer[2] = channel
+        self._data_buffer[3] = self._sequence_number[channel]
+        for idx, send_byte in enumerate(data):
+            self._data_buffer[4 + idx] = send_byte
```

Comment:
```suggestion
            self._data_buffer[4:] = data
```

Reply:
@siddacious please try the change since you're in test mode

Reply:
@ladyada tested and working with a minor change so that `self._data_buffer` retains its length